apps/visitor-app/backend/service.bal

```diff
@@ -255,4 +261,330 @@ service http:InterceptableService / on new http:Listener(9090) {
+                }
+            };
+        }
+
+        error? emailError = email:sendEmail({
+                                                to: [payload.inviteeEmail],
+                                                'from: email:visitorNotificationFrom,
+                                                subject: email:VISIT_INVITATION_SUBJECT,
+                                                template: content,
+                                                cc: [email:receptionEmail]
+                                            });
+
+        if emailError is error {
+            string errMsg = "Error occurred while sending the email!";
+            log:printError(errMsg, emailError);
+        }
+
+        return <http:Created>{
+            body: {
+                message: "Invitation created successfully!"
+            }
+        };
+    }
+
+    # Fetch invitation details using the encoded value.
+    #
+    # + encodeValue - Encoded value from the invitation link
+    # + return - Invitation details or error
+    resource function get invitations/[string encodeValue]/authorize() returns database:Invitation|http:InternalServerError {
+        database:Invitation|error invitationDetails = database:checkInvitation(encodeValue);
+
+        if invitationDetails is error {
+            string errMsg = invitationDetails.message();
+            return <http:InternalServerError>{
+                body: {
+                    message: errMsg
+                }
+            };
+        }
+
+        database:VisitsResponse|error visitsResponse = database:fetchVisits(invitation_id = invitationDetails.invitationId);
+        if visitsResponse is error {
+            string errMsg = "Error occurred while fetching visits!";
+            log:printError(errMsg, visitsResponse);
+            return <http:InternalServerError>{
+                body: {
+                    message: errMsg
+                }
+            };
+        }
+
+        database:AddVisitorPayload[] inviteesList = from database:Visit visit in visitsResponse.visits
+            select {
+                nicHash: visit.nicHash,
+                name: visit.name,
+                email: visit.email,
+                contactNumber: visit.contactNumber,
+                nicNumber: visit.nicNumber
+            };
+
+        invitationDetails.invitees = inviteesList;
+        return invitationDetails;
+    };
+
+    # Fill an invitation by adding a visitor and a visit.
+    #
+    # + encodeValue - Encoded value from the invitation link
+    # + payload - Payload containing the visitor details
+    # + return - Successfully created or error
+    resource function post invitations/[string encodeValue]/fill(database:AddVisitorPayload payload)
+        returns http:Created|http:BadRequest|http:InternalServerError|error? {
+
+        database:Invitation|error invitationDetails = database:checkInvitation(encodeValue);
+
+        if invitationDetails is error {
```

Comment:
use the following approach
string errMsg = "Error occurred while fetching visits!";
            log:printError(errMsg, visitsResponse);
            return <http:InternalServerError>{
                body: {
                    message: errMsg
                }
            };